backend/modules/simulation/service.py

```
@@ -225,9 +225,11 @@ async def stream_chat_message(
         acquired = await acquire_stream_slot()
         if not acquired:
             # Immediate back-pressure: inform client that capacity is reached.
+            logger.warning(f"[CAPACITY] Stream slot unavailable for user {user_id}, user_progress_id {user_progress_id} - system at capacity")
```

Ruff (0.14.8)

Undefined name `logger`

(F821)